tablepyxl/style.py

```diff
@@ -252,7 +252,7 @@ def data_type(self):
             if 'TYPE_FORMULA' in cell_types:
                 # Make sure TYPE_FORMULA takes precedence over the other classes in the set.
                 cell_type = 'TYPE_FORMULA'
-            elif 'TYPE_CURRENCY' in cell_types or 'TYPE_INTEGER' in cell_types:
+            elif 'TYPE_CURRENCY' in cell_types or 'TYPE_INTEGER' in cell_types or 'TYPE_PERCENTAGE' in cell_types:
```

Comment:
This looks like its getting a bit unwieldy, consider doing something like: 
```
numeric_types =  ['TYPE_CURRENCY', 'TYPE_INTEGER', 'TYPE_PERCENTAGE']
elif any(type in cell_types for type in numeric_types):
```

Reply:
I wonder about where to put the definition of numeric_types. If it's too far away from this code, you have to navigate around to see if the type you're interested in is in the list.

cell_types is a set, so we could do:
```
elif cell_types & {'TYPE_CURRENCY', 'TYPE_INTEGER', 'TYPE_PERCENTAGE'}:
```

Whatever you do, don't use `type` as an identifier since it shadows a builtin.

Reply:
briefly contemplated breaking out the definition of CELL_TYPES into
```
NUMERIC_CELL_TYPES = {'TYPE_INTEGER', 'TYPE_CURRENCY', 'TYPE_PERCENTAGE'}
OTHER_CELL_TYPES = {'TYPE_whatever else there is'}
CELL_TYPES = NUMERIC_CELL_TYPES + OTHER_CELL_TYPES
```
so we could do `elif cell_types & self.NUMERIC_TYPES:` but for now, I am going with Brad's suggestion.